pkg/pool/pool.go

```diff
@@ -102,8 +85,10 @@ type pool struct {
 
 func newPool(ctx context.Context, options *BuilderOptions) (Pool, error) {
 	clientPacks := make([]*clientPack, len(options.weights))
-	for i, con := range options.connections {
-		c, err := client.New(client.WithDefaultPrivateKey(options.Key), client.WithGRPCConnection(con))
+	for i, node := range options.nodes {
+		c, err := client.New(client.WithDefaultPrivateKey(options.Key),
+			client.WithURIAddress(node.address, node.tlsCfg),
```

Comment:
From what I've seen in nspcc-dev/neo-go#2003, we can pass `grpcs://` address with `nil` TLS configuration and it'll work with the default TLS settings which is fine for us (you're using `&tls.Config{}` anyway). So maybe we can skip address parsing completely in sdk-go?

Reply:
If we always use the default tls config in services (http-gw, s3-gw), we can skip parsing here